common.mk

```diff
@@ -0,0 +1,102 @@
+#
+# Common definition to all platforms
+#
+
+BASH := $(shell which bash)
+ROOT ?= $(shell pwd)/..
+
+#
+# Must declare in platform specific makefiles:
+# - CROSS_COMPILE_NS_USER / CROSS_COMPILE_NS_KERNEL
+# - CROSS_COMPILE_S_USER / CROSS_COMPILE_S_KERNEL
+#
+# - OPTEE_OS_BIN
+# - OPTEE_OS_TA_DEV_KIT_DIR
+#
+
+LINUX_PATH			?= $(ROOT)/linux
+OPTEE_OS_PATH			?= $(ROOT)/optee_os
+OPTEE_CLIENT_PATH		?= $(ROOT)/optee_client
+OPTEE_CLIENT_EXPORT		?= $(OPTEE_CLIENT_PATH)/out/export
+OPTEE_LINUXDRIVER_PATH		?= $(ROOT)/optee_linuxdriver
+OPTEE_TEST_PATH			?= $(ROOT)/optee_test
+OPTEE_TEST_OUT_PATH 		?= $(ROOT)/optee_test/out
+
+CCACHE ?= $(shell which ccache) # Don't remove this comment (space is needed)
```

Comment:
No for this one. In case ccache is **not** installed, then `which ccache` will not output anything, and this will not lead to an error.